src/lib.rs

```diff
@@ -51,40 +52,58 @@ const TEMP_DIR_NAME: &str = "aoc_cache";
 /// # Example
 ///
 /// ```
-/// use aoc_cache::get_input_from_web_or_cache;
+/// use aoc_cache::get;
 /// // my.cookie is a file containing the cookie string.
 /// const MY_COOKIE: &str = include_str!("my.cookie");
 /// let input: Result<String, aoc_cache::Error> =
-///     get_input_from_web_or_cache("https://adventofcode.com/2022/day/1/input", MY_COOKIE);
+///     get("https://adventofcode.com/2022/day/1/input", MY_COOKIE);
 /// ```
 ///
 /// [github-cookie-example]: https://github.com/wimglenn/advent-of-code-wim/issues/1
 /// [google-cookie]: https://www.google.com/search?q=adventofcode+cookie
 #[instrument(skip(cookie))]
-pub fn get_input_from_web_or_cache(url: &str, cookie: &str) -> Result<String> {
+pub fn get(url: &str, cookie: &str) -> Result<String> {
     if let Some(content) = get_cache_for_url(url)? {
         info!("returning content found in cache");
         return Ok(content);
     }
     debug!("content not found in cache, requesting from web");
+    let content = get_content_from_web(url, cookie)?;
+    add_cache(url, &content)?;
+    info!("returning content from web");
+    Ok(content)
+}
+
+/// Dispatches to the `get`-function, see its documentation.
+#[deprecated]
+pub fn get_input_from_web_or_cache(url: &str, cookie: &str) -> Result<String> {
+    get(url, cookie)
+}
+
+#[instrument(skip(url, cookie))]
+fn get_content_from_web(url: &str, cookie: &str) -> Result<String> {
     if cookie.is_empty() {
         return Err(Error::InvalidCookie(
             "empty cookie is not valid".to_string(),
         ));
     }
-    let jar = Jar::default();
+
     let url_parsed = url.parse()?;
-    jar.add_cookie_str(cookie, &url_parsed);
-    let client = Client::builder()
-        .cookie_store(true)
-        .cookie_provider(Arc::new(jar))
-        .user_agent("https://github.com/glennib/aoc-cache by glennib.pub@gmail.com")
-        .build()?;
-    let request = client.get(url_parsed).build()?;
-    let response = client.execute(request)?.error_for_status()?;
-    let content = response.text()?.trim().to_string();
-    add_cache(url, &content)?;
-    info!("returning content from web");
+
+    let jar = CookieStore::load(BufReader::new(cookie.as_bytes()), |s| {
+        trace!(s, "parsed cookie from str");
+        Cookie::parse(s, &url_parsed).map(Cookie::into_owned)
+    })
+    .map_err(|_| Error::CookieParse("couldn't create cookie store".into()))?;
+
+    debug!(?jar);
+
+    let agent = AgentBuilder::new()
+        .cookie_store(jar)
+        .user_agent(USER_AGENT)
+        .build();
+    let response = agent.get(url).call()?;
+    let content = response.into_string()?.trim().to_string();
```

Comment:
Looking at the implementation of ureq it appears that [non-200 status are not a `Result::Err`](https://docs.rs/ureq/latest/src/ureq/response.rs.html#599) and we're not checking the status before reading the content. I think we still want to check the status and only accept the response content if its 200 for the following reasons:

* cookie may parse as valid, but may be expired or otherwise non-functional. in this case you may still get a 400
* the user may try to access a puzzle input too early, where they get a 404 (I accidentally did this myself apparently)

If the status is not 200 the content is almost certainly invalid. This forces the user to `cargo clean` which may result in re-downloading inputs for *all* days, increasing the load on aoc servers and subverting the purpose of this package.